daemon/network.go

```diff
@@ -515,43 +471,16 @@ func (daemon *Daemon) DeleteNetwork(networkID string) error {
 	return daemon.deleteNetwork(networkID, false)
 }
 
-func (daemon *Daemon) deleteLoadBalancerSandbox(n libnetwork.Network) {
-	controller := daemon.netController
-
-	//The only endpoint left should be the LB endpoint (nw.Name() + "-endpoint")
-	endpoints := n.Endpoints()
-	if len(endpoints) == 1 {
-		sandboxName := n.Name() + "-sbox"
-
-		info := endpoints[0].Info()
-		if info != nil {
-			sb := info.Sandbox()
-			if sb != nil {
-				if err := sb.DisableService(); err != nil {
-					logrus.Warnf("Failed to disable service on sandbox %s: %v", sandboxName, err)
-					//Ignore error and attempt to delete the load balancer endpoint
-				}
-			}
-		}
-
-		if err := endpoints[0].Delete(true); err != nil {
-			logrus.Warnf("Failed to delete endpoint %s (%s) in %s: %v", endpoints[0].Name(), endpoints[0].ID(), sandboxName, err)
-			//Ignore error and attempt to delete the sandbox.
-		}
-
-		if err := controller.SandboxDestroy(sandboxName); err != nil {
-			logrus.Warnf("Failed to delete %s sandbox: %v", sandboxName, err)
-			//Ignore error and attempt to delete the network.
-		}
-	}
-}
-
 func (daemon *Daemon) deleteNetwork(networkID string, dynamic bool) error {
 	nw, err := daemon.FindNetwork(networkID)
 	if err != nil {
 		return err
 	}
 
+	if nw.Info().Ingress() {
```

Comment:
Can you pls explain why we need this ?

Reply:
Design wise, I treated ingress like other overlay networks with a load balancer sandbox.
It requires the same steps to setup and remove the network (i.e. we must create the ingress-sbox, the ingress-endpoint, etc).
However, the ingress sandbox's lifetime is slightly different than the other networks.
The ingress sandbox exists even when there are no tasks connected to it.
But, other networks create the load balancer sandbox on the fly when the first task is scheduled, and remove it when the last task goes away.

Currently, when a task is removed, we delete all the networks it is connect to.  
For ingress, this always returns "ActiveEndpoints" error.
The ingress sandbox and network are actually deleted in releaseIngress().

Now, by moving the deletion of the lb sandbox into libnetwork, libnetwork needs to figure out when to delete the lb sandbox (https://github.com/docker/libnetwork/pull/2011/files#diff-cba6b1b3488fe48ebbef7e1821a3745dR962).  Libnetwork, deletes the network when the only endpoint left is the lb endpoint.  
But, the ingress sandbox's exists even when there are no tasks left.  So, we don't delete the ingress network here and only in releaseIngress().

One alternative I considered was to make the ingress's sandbox lifetime behave exactly like other overlay sandbox's.  I.e. only create the ingress sandbox when a task is scheduled and remove it when the last task is gone.  That would remove these special cases for ingress.  However, I felt like that would be a bigger design change with more ramifications.

Reply:
The suggestion to change the ingress sandbox lifetime wont work, because the purpose of ingress network was to provide the overlay connectivity from any node for any service with a task attached to any other node.

Also, technically the `ingress-sbox` is similar to the `lb-sandbox` that you have introduced. But the management of this sandbox and the ingress-endpoint is handled outside of your existing changes (I may have to spend some more time to understand if that is not the case). PS: I didnt review your previous changes and that might answer the lack of my knowledge in this area of the code.
I will comment more on this once I review the `ingress-sbox` management more thoroughly.

Reply:
@pradipd okay. the changes makes sense.